src/states/node.rs

```diff
@@ -1579,7 +1561,13 @@ impl Node {
                    route: u8,
                    sent_to: &[XorName])
                    -> Result<(Vec<XorName>, Vec<PeerId>), RoutingError> {
-        if self.is_proper() {
+        let force_via_proxy = match routing_msg.content {
```

Comment:
Good catch.  I should be checking `routing_msg.src` as well so `force_via_proxy` is true where we're sending as a Client and the message is from us (i.e. we're not acting as a proxy ourself).